### ISSUE TITLE
Any time we take a profile GUID, consider taking a profile name

### ISSUE THREAD

**Issue by DHowett-MSFT** [microsoft/terminal#3680]
starts with `{` = guid, and we'll need a helper that looks in your profile list for a name and turns it into a guid. This'll allow people to specify guidless profiles in future key bindings that let you specify guids, as well as in `defaultProfile`

**Comment by DHowett-MSFT** [comment]
If somebody names their profile `{something that looks like a guid}`, I mean, that's their own fault.

**Comment by zadjii-msft** [comment]
> 
> 
> If somebody names their profile `{something that looks like a guid}`, I mean, that's their own fault.

If we fail to parse a GUID, we can always just fall back to looking up a name. Though I suppose if someone names their profile `{an-actual-guid}`, that's just begging for things to not work right.

**Comment by mkitzan** [comment]
@DHowett-MSFT I can add a helper function in `CascadiaSettings` which takes a `wstring` profile name and returns a `GUID`, if there's a match. Is this kind of what you're thinking with this one?

**Comment by DHowett-MSFT** [comment]
@mkitzan yep! I'd be careful though: @zadjii-msft is probably going to be working in this area _soon_, but you might beat him to it :smile:

**Comment by mkitzan** [comment]
@DHowett I've got the helper implemented which I can open a PR for. However, when looking at `Profile`, I found a function [`GetGuid`](https://github.com/microsoft/terminal/blob/master/src/cascadia/TerminalApp/Profile.cpp#L141) marked `noexcept` which throws if the GUID is illegitimate. Many `noexcept` functions in `CascadiaSettings` use this function. Could be a bad time if this were to throw. Is the dev team aware of this/purposely did this? If not, I can open an issue and fix the `noexcept` labels (or just make the fix in the PR for this issue).
